lib/iris/tests/unit/concatenate/test_concatenate.py

```diff
@@ -246,6 +246,45 @@ def test_dim_coords_overlap_message(self):
             _ = concatenate([cube_1, cube_2], True)
 
 
+class TestNonMetadataMessages(TestMessages):
```

Comment:
This unfortunately inherits a lot of other tests, too, so those are all running twice. The correct thing to do here would be to:

- create a `_MessagesMixin` class that ONLY has the `setUp()` method
- have `TestMessages` inherit from `_MessagesMixin`
- have your new class also inherit from `_MessagesMixin`, leaving you free to write whatever tests you want with whatever names you want